### ISSUE TITLE
grass 70 build fails

### ISSUE THREAD

**Issue by mbalcilar** [OSGeo/homebrew-osgeo4mac#118]
Logs are here: https://gist.github.com/eb8beaad519dfb5ccdf0

**Comment by dakcarto** [comment]
Hi,

Thanks for reporting! Please try latest formula that install 7.0.3. Tested on 10.11.

**Comment by rkrug** [comment]
Which OS? all grass do not compile on El Capitan - unless you disable SIP for installation. Afterwards, you can turn it on again.

**Comment by dakcarto** [comment]
@rkrug, please see latest [grass-70.rb commit](https://github.com/OSGeo/homebrew-osgeo4mac/commit/726cb94), where I got 7.0.3 to compile on OS X 10.11 with SIP enabled.

If you have time, please test the compilation.

**Comment by rkrug** [comment]
Yes it does install - This is BRILLIANT news. did you backport the changes to grass-64 as well?

**Comment by rkrug** [comment]
Just announced your success on the grass lists - hope y=this is OK.

**Comment by mankoff** [comment]
I'm excited by this news, but it does not work for me. I'm on OS X 10.11.3 and the compilation fails. Should I post all logs? Just the end of one specific log?

**Comment by rkrug** [comment]
10.11.4 worked for me - didn't try it before.
Easiest should be to do the following (from https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting):

> Upload debugging information to a Gist:
> 
>     If you had a formula error: run `brew gist-logs <formula> ` (where <formula> is the name of the formula that failed to build).
>     If you encountered a non-formula bug: upload the output of brew config and brew doctor to a new Gist.
> 

provide here the url to the gist.

**Comment by mankoff** [comment]
Have you installed the framework packages, or is this on a "clean" system and using only Brew-installed packages? I'm trying to do this w/ only Brew.

**Comment by mankoff** [comment]
Well `brew gist-logs` crashes on me due to (known?) `brew` bug discussed here https://github.com/Homebrew/homebrew/issues/50180 and https://github.com/Homebrew/homebrew/pull/49846.

**Comment by dakcarto** [comment]
@mankoff, by "framework packages" do you mean from Kyngchaos.com? No, I only use Homebrew on this particular dev box.

Here is my config under which I built grass-70 (7.0.3), for reference:
```
$ brew config
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Homebrew/homebrew
HEAD: ce1d3f3f3528156c624c8ebdaf54eea6b06edf01
Last commit: 7 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: 8-core 64-bit haswell
OS X: 10.11.3-x86_64
Xcode: 7.2.1
CLT: 7.2.0.0.1.1447826929
Clang: 7.0 build 700
X11: 2.7.8 => /opt/X11
System Ruby: 2.0.0-p645
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_74, 1.7.0_79
```
So, a little outdated on the Xcode and OS updates.

**Comment by dakcarto** [comment]
@rkrug wrote:
> Yes it does install - This is BRILLIANT news. did you backport the changes to grass-64 as well?

No, I have not backported anything. Pretty busy at work. Please give it a go and post a PR if it works. Thanks!

**Comment by mankoff** [comment]
Yes I meant Kyngchaos.com. I have a similar setup:

```
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Homebrew/homebrew
HEAD: 54e3ba354514dc85f6cb860255b309143a986073
Last commit: 5 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit broadwell
OS X: 10.11.3-x86_64
Xcode: 7.2.1
CLT: 7.3.0.0.1.1457485338
Clang: 7.0 build 700
X11: 2.7.8 => /opt/X11
System Ruby: 2.0.0-p645
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.3.0/bin/ruby
Java: 1.8.0_45, 1.6.0_65-b14-468
```

And when I run ` brew install grass-70 --without-gui` I get this:

```
Hunk #1 succeeded at 116 (offset 2 lines).
==> ./configure --prefix=/usr/local/Cellar/grass-70/7.0.3 --enable-shared --with-cxx --with-python --with-blas --with-
==> make GDAL_DYNAMIC=
Last 15 lines from /Users/mankoff/Library/Logs/Homebrew/grass-70/02.make:
/private/tmp/grass-7020160323-59511-1ql7nqq/grass-7.0.3/temporal/t.vect.db.select
/private/tmp/grass-7020160323-59511-1ql7nqq/grass-7.0.3/temporal/t.vect.export
/private/tmp/grass-7020160323-59511-1ql7nqq/grass-7.0.3/temporal/t.vect.extract
/private/tmp/grass-7020160323-59511-1ql7nqq/grass-7.0.3/temporal/t.vect.import
/private/tmp/grass-7020160323-59511-1ql7nqq/grass-7.0.3/temporal/t.vect.what.strds
/private/tmp/grass-7020160323-59511-1ql7nqq/grass-7.0.3/temporal/t.vect.univar
/private/tmp/grass-7020160323-59511-1ql7nqq/grass-7.0.3/temporal/t.vect.observe.strds
--
In case of errors please change into the directory with error and run 'make'.
If you get multiple errors, you need to deal with them in the order they
appear in the error log. If you get an error building a library, you will
also get errors from anything which uses the library.
--
Finished compilation: Wed Mar 23 12:46:11 EDT 2016
make: *** [default] Error 1

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/homebrew):
  https://github.com/osgeo/homebrew-osgeo4mac/issues
```

**Comment by rkrug** [comment]
I had the t.vect errors before as well - but they were gne now.

Try:

`brew install --debug grass-70 --without-gui`

This will give you the option to go into the sheel when an error occurs while compiling and you can do as GRASS suggests: going into the directory where the first error occurred and call `make` to see why thaere is an error.


Also pure homebrew - took me some time to get rid of the frame
works.

**Comment by mankoff** [comment]
I've never installed the frameworks!  It is a python library-not-found error that causes the crash. It is looking for `/usr/local/Cellar/grass-70/7.0.3/grass-7.0.3/lib/libgrass_datetime.7.0.3.dylib` but `/usr/local/Cellar/grass-70/7.0.3/` is an empty folder (probably because `grass` hasn't been installed yet).

```
Traceback (most recent call last):
  File "/private/tmp/grass-7020160323-96132-460ghb/grass-7.0.3/dist.x86_64-apple-darwin15.3.0/scripts/t.support", line 71, in <module>
    import grass.temporal as tgis
  File "/private/tmp/grass-7020160323-96132-460ghb/grass-7.0.3/dist.x86_64-apple-darwin15.3.0/etc/python/grass/temporal/__init__.py", line 1, in <module>
    from core import *
  File "/private/tmp/grass-7020160323-96132-460ghb/grass-7.0.3/dist.x86_64-apple-darwin15.3.0/etc/python/grass/temporal/core.py", line 38, in <module>
    from c_libraries_interface import *
  File "/private/tmp/grass-7020160323-96132-460ghb/grass-7.0.3/dist.x86_64-apple-darwin15.3.0/etc/python/grass/temporal/c_libraries_interface.py", line 19, in <module>
    import grass.lib.gis as libgis
  File "/private/tmp/grass-7020160323-96132-460ghb/grass-7.0.3/dist.x86_64-apple-darwin15.3.0/etc/python/grass/lib/gis.py", line 23, in <module>
    _libs["grass_gis.7.0.3"] = load_library("grass_gis.7.0.3")
  File "/private/tmp/grass-7020160323-96132-460ghb/grass-7.0.3/dist.x86_64-apple-darwin15.3.0/etc/python/grass/lib/ctypes_loader.py", line 55, in load_library
    return self.load(path)
  File "/private/tmp/grass-7020160323-96132-460ghb/grass-7.0.3/dist.x86_64-apple-darwin15.3.0/etc/python/grass/lib/ctypes_loader.py", line 71, in load
    raise ImportError,e
ImportError: dlopen(/private/tmp/grass-7020160323-96132-460ghb/grass-7.0.3/.brew_home/lib/libgrass_gis.7.0.3.dylib, 10): Library not loaded: /usr/local/Cellar/grass-70/7.0.3/grass-7.0.3/lib/libgrass_datetime.7.0.3.dylib
  Referenced from: /private/tmp/grass-7020160323-96132-460ghb/grass-7.0.3/.brew_home/lib/libgrass_gis.7.0.3.dylib
  Reason: image not found
```

**Comment by rkrug** [comment]
My config:
```

3:12:47 ~$ brew config
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Homebrew/homebrew.git
HEAD: 54e3ba354514dc85f6cb860255b309143a986073
Last commit: 6 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: 8-core 64-bit ivybridge
OS X: 10.11.4-x86_64
Xcode: 7.3
CLT: N/A
GCC-4.2: build 5666
Clang: 7.3 build 703
X11: 2.7.8 => /opt/X11
System Ruby: 2.0.0-p648
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.3.0/bin/ruby
Java: 1.8.0_20
06:20:28 ~$
```

The only difference I see is python.

**Comment by mankoff** [comment]
Thanks for the hint. I did `brew install python` and then had to `brew install numpy` (or should it be `pip install`?) and then... it works! Thanks for the help!

**Comment by rkrug** [comment]
Hm - this should than be in the dependencies of grass-70 - is this possible?

**Comment by mankoff** [comment]
One more note here... I have Anaconda python3 installed and set as my 'system' python. I've tested this by installing the Anaconda package, or `brew install python3` and then `pip3 install conda`. Either case conflicts with grass. I've solved this by changing the top line of `/usr/local/bin/grass70` from `#!/usr/bin/env python` to `#!/usr/bin/env python2.7`

Now `python` brings up my python3 install, and `grass70` still works. This should be moot soon once the grass python3 port is complete.
